smart-contracts/contracts/burn_coins/src/query.rs

```diff
@@ -0,0 +1,103 @@
+use cosmwasm_std::{Coin, Deps, Order, StdResult};
+
+use crate::msg::TotalBurntResponse;
+use crate::state::AMOUNT_BURNT;
+
+pub fn query_total_burn(deps: Deps) -> StdResult<TotalBurntResponse> {
+    let mut total_burn_amount: Vec<Coin> = vec![];
+
+    for res in AMOUNT_BURNT.range(deps.storage, None, None, Order::Ascending) {
```

Comment:
nitpick: this can be a bit cleaner

this can also be done by collecting the range in a vec, something like:
```
let total_burn_amount: Result<Vec<Coin>, ContractError> = AMOUNT_BURNT.range(deps.storage, None, None, Order::Ascending).collect
```
Depending on the contents of AMOUNT_BURNT, you might need to do some more work through `.map()`, such as wrapping the key and value in a coin